flocker/common/_era.py

```diff
@@ -0,0 +1,65 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Era information for Flocker nodes.
+
+Every time a node reboots it gets a new, globally unique era.
+"""
+
+import sys
+from uuid import UUID
+
+from zope.interface import implementer
+
+from twisted.internet.defer import succeed
+from twisted.python.filepath import FilePath
+from twisted.python.usage import Options
+
+from ..common.script import (
+    ICommandLineScript, flocker_standard_options, FlockerScriptRunner,
+)
+
+
+_BOOT_ID = FilePath(b"/proc/sys/kernel/random/boot_id")
+
+
+def get_era():
+    """
+    :return UUID: A node- and boot-specific globally unique id.
+    """
+    return UUID(hex=_BOOT_ID.getContent().strip())
+
+
+@flocker_standard_options
+class EraOptions(Options):
+    """
+    Command line options for ``flocker-node-era``.
+    """
+    longdesc = (
+        "Print the current node's era to stdout. The era is a unique"
+        "identifier per reboot per node, and can be used to discover the"
+        "current node's state safely using Flocker's REST API.\n"
```

Comment:
Nit: I think "safely" raises more questions than it answers, and should thus be removed from the description.